.github/workflows/linux.yml

```diff
@@ -38,3 +38,52 @@ jobs:
         env:
           MAX_WARNINGS: ${{ matrix.conf.max_warnings }}
         run:  ./scripts/count-warnings.py build.log
+
+  build_linux_release_dynamic:
+    name: Release build (dynamic)
+    runs-on: ubuntu-16.04
+    steps:
+      - uses: actions/checkout@v1
+      - run:  sudo apt-get update
+      - name: Install C++ compiler and libraries
+        run:  sudo apt-get install -y tree $(./scripts/list-build-dependencies.sh -m apt -c gcc)
+      - name: Log environment
+        run:  ./scripts/log-env.sh
+      - name: Build
+        run: |
+          set -x
+          export VERSION=$(git describe --abbrev=4)
+          sed -i "s/AC_INIT(dosbox,git)/AC_INIT(dosbox,$VERSION)/" configure.ac
```

Comment:
I really like what this does:

![2019-12-05_07-30](https://user-images.githubusercontent.com/1557255/70249839-d6096e00-1731-11ea-9b0f-5342eaf275e9.png)

But the flip side is this:
```
[kcroft@hoya tmp] ll ~/.dosbox/dosbox*
-rw-rw---- 1 kcroft users 11K Apr  1  2018 /home/kcroft/.dosbox/dosbox-0.74.conf
lrwxrwxrwx 1 kcroft users  16 Jun 12  2016 /home/kcroft/.dosbox/dosbox.conf -> dosbox-0.74.conf
-rw-r--r-- 1 kcroft users 18K Jun 15  2017 /home/kcroft/.dosbox/dosbox-ECE.conf
lrwxrwxrwx 1 kcroft users  15 Oct  6 01:15 /home/kcroft/.dosbox/dosbox-git.conf -> dosbox-SVN.conf
-rw-rw---- 1 kcroft users 11K Nov 29 16:36 /home/kcroft/.dosbox/dosbox-SVN.conf
-rw-r--r-- 1 kcroft users 13K Dec  5 07:31 /home/kcroft/.dosbox/dosbox-v0.75.0-pre-196-gb7377.conf
-rw-r--r-- 1 kcroft users 13K Dec  5 07:31 /home/kcroft/.dosbox/dosbox-v0.75.0-pre-197-g2bb6.conf
-rw-r--r-- 1 kcroft users 13K Dec  5 07:31 /home/kcroft/.dosbox/dosbox-v0.75.0-pre-198-ga4d5e.conf
-rw-r--r-- 1 kcroft users 13K Dec  5 07:28 /home/kcroft/.dosbox/dosbox-v0.75.0-pre-201-ge847.conf
-rw-r--r-- 1 kcroft users 13K Dec  5 07:32 /home/kcroft/.dosbox/dosbox-v0.75.0-pre-202-ge913e.conf
```

The pros of unique conf files is we can add or remove features and those get reflected in the newly-generated conf, which the user can tweak and independently test without touching other conf files.  Technically this is the most robust and "correct" approach. 

The cons are that this is too much for us as people to really deal with.  Even with these 5 examples, the wind has left my sail with any desire to go re-edit my prefered default settings into any of them; knowing that they are now so cheaply generated for every build.  I simply give up. 

The quantity of confs really builds up fast too - and eventually I will have to cleanout my `~/.dosbox` but be careful not to accidentally delete my couple prefered .conf files that I've hand-adjusted for quite a while. 

Some options:
- Step back the version used to read to conf, so it's at a higher level for example, if all of these new builds simply used `.dosbox/dosbox-v0.75.0.conf`, maybe I would feel compelled to deal with that. 
- Go even simpler: make them all keep using `.dosbox/dosbox-git.conf` and we mention any conf changes in the README, Wiki, or some other announcement page.  I would prefer this, as a user.  I just don't like the idea of managing even more conf files.
- Do away with versioned confs all together, just make it use `~/.dosbox/dosbox.conf` or `~/.dosbox/dosbox-staging.conf` forever-more; then there are no surprises when upgrading (which could be every ~3 to 6 months).  This is probably my most prefered setting, if I were to "fast-forward" my life and dealing with `.dosbox/dosbox-v0.75.1.conf`, `.dosbox/dosbox-v0.75.2.conf`, then `.dosbox/dosbox-v0.76.0.conf`, and `.dosbox/dosbox-v0.76.1.conf`. and so on.. I will probably get sick of that too!)

Whatever our choice, unless we stick with the default we should separate the version shown in the UI versus the value stamped on the conf file (ideally just `dosbox-staging.conf` or `dosbox-git.conf` and done.. ).

Reply:
Ah, right - configuration files are versioned in DOSBox :( I agree, this is too much. It's not normal to append a version name to user configuration file. I will try to keep version included in `--version` and on intro banner.

Regarding the name of config file - not sure yet. Maybe `dosbox-staging-git.conf` for development and "snapshot" builds and reserve `dosbox-staging.conf` for proper release builds? Overall, current way of handling .conf files is pretty… broken. But fixing it will require a whole new feature to carefully consider all user scenarios, consider cross-platform support, etc.

Reply:
>Maybe dosbox-staging-git.conf for development and "snapshot" builds and reserve dosbox-staging.conf

Sounds perfectly reasonable; I like this and could certainly live with it.